### ISSUE TITLE
[1.0.0] Test failure on mips and riscv64 (integer overflow)

### ISSUE THREAD

**Issue by olebole** [astropy/photutils#1085]
When building the new version for MIPS, I observe a failure in the tests that I don't fully understand:

```
    def test_epsfmodel_inputs():
        data = np.array([[], []])
        with pytest.raises(ValueError):
            EPSFModel(data)
    
        data = np.ones((5, 5), dtype=float)
        data[2, 2] = np.inf
        with pytest.raises(ValueError):
            EPSFModel(data)
    
        data[2, 2] = np.finfo(float).max * 2
        with pytest.raises(ValueError):
            EPSFModel(data, flux=None)
    
        data[2, 2] = 1
        for oversampling in [np.NaN, 'a', -1, [-2, 4]]:
            with pytest.raises(ValueError):
>               EPSFModel(data, oversampling=oversampling)

photutils/psf/tests/test_epsf.py:181: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
photutils/psf/models.py:548: in __init__
    super().__init__(data=data, flux=flux, x_0=x_0, y_0=y_0,
photutils/psf/models.py:157: in __init__
    flux = self._initial_norm(flux, normalize)
photutils/psf/models.py:561: in _initial_norm
    self._compute_normalization()
photutils/psf/models.py:592: in _compute_normalization
    self._img_norm = self._compute_raw_image_norm()
photutils/psf/models.py:577: in _compute_raw_image_norm
    flux, _ = aper.do_photometry(self._data, method='exact')
photutils/aperture/core.py:388: in do_photometry
    return self._do_photometry(data, variance, method=method,
photutils/aperture/core.py:286: in _do_photometry
    masks = self.to_mask(method=method, subpixels=subpixels)
photutils/aperture/circle.py:86: in to_mask
    mask = circular_overlap_grid(edges[0], edges[1], edges[2],
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

>   ???
E   OverflowError: Python int too large to convert to C long

photutils/geometry/circular_overlap.pyx:31: OverflowError
```
This happens on [MIPS 32 bit](https://buildd.debian.org/status/fetch.php?pkg=photutils&arch=mipsel&ver=1.0.0-1&stamp=1600856942&raw=0), [MIPS 64 bit](https://buildd.debian.org/status/fetch.php?pkg=photutils&arch=mips64el&ver=1.0.0-1&stamp=1600856316&raw=0), [RISCV64](https://buildd.debian.org/status/fetch.php?pkg=photutils&arch=riscv64&ver=1.0.0-1&stamp=1600856959&raw=0), and [HP-PA](https://buildd.debian.org/status/fetch.php?pkg=photutils&arch=hppa&ver=1.0.0-1&stamp=1600857995&raw=0).
Other platforms (amd64, i386, arm, arm64 etc.) [build nicely](https://buildd.debian.org/status/logs.php?pkg=photutils&ver=1.0.0-1&suite=sid).

**Comment by larrybradley** [comment]
Many thanks, @olebole.  Seems like an odd error since I would expect the `ValueError` to be raised immediately (that must not be happening).  I highly suspect that if `np.NaN` is removed from the list (`[np.NaN, 'a', -1, [-2, 4]]`) that it will pass.  Is it possible for you to make that small change and see if that makes your tests pass?

**Comment by larrybradley** [comment]
Fixed by #1086.
